domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -492,16 +492,37 @@ class TopicListController @Inject constructor(
     val impliedFinishedTopicIds = computeImpliedCompletedDependencies(
       fullyCompletedTopicIds, topicDependencyMap
     )
-    // Suggest prerequisite topic user needs to learn after completing any of the topics.
+    // Suggest prerequisite topic user needs to learn while user is learning any of the topics.
     for (topicId in unstartedTopicIdList) {
-      // All of the topic's prerequisites must be completed before it can be suggested.
+      // All of the topic's prerequisites can be suggested if the topic is ongoing.
       val dependentTopicIds = topicDependencyMap[topicId] ?: listOf()
-      if (topicId !in impliedFinishedTopicIds && startedTopicIds.containsAll(dependentTopicIds!!)) {
+      if (topicId !in impliedFinishedTopicIds && startedTopicIds.any { it in dependentTopicIds } &&
```

Comment:
Nice catch on not using startedTopicIds, but shouldn't this be:

``if (topicId !in impliedFinishedTopicIds && impliedFinishedTopicIds.containsAll(dependentTopicIds))``

This reads as: if the topic itself has not been completed, and if no later topic that depends (directly or indirectly) on this topic has not been completed, and if all of the dependencies of this topic can be considered completed (either they were directly completed, or topics that depend on them directly/indirectly were completed) then we can recommend the topic.

That *seems* correct, and comprehensive. Are there any cases missing from this?

As I understand it, this should also work in the case where the learner completes the last topic in the graph & we still suggest other topics that have the same prerequisite dependencies (similar to the example I gave in the email thread). Can you make sure we have a test covering that exact case, too?

Reply:
`impliedFinishedTopicIds.containsAll(dependentTopicIds)` returns false. Made use of `any()` to check at least dependent Id is present.